apps/src/Sounds.js

```diff
@@ -235,6 +240,48 @@ Sounds.prototype.checkDidSourcePlay_ = function(source, context, onComplete) {
   );
 };
 
+/**
+ * Registers the bytes of text to speech.
+ * @param {string} language the language the text is spoken in.
+ * @param {string} text the text that is being spoken.
+ * @param {boolean} hasProfanity
+ * @param {Array.<string>} profaneWords an array of profane words in the text if there are any. Otherwise null.
+ * @param {string} gender the gender of the voice used to speak the text. Optional
+ * @param {Array.<bytes>} bytes the bytes of the speech.
+ */
+Sounds.prototype.registerTextBytes = function(
+  language,
+  text,
+  hasProfanity,
+  profaneWords = undefined,
+  gender = undefined,
+  bytes = undefined
+) {
+  if (!this.textBytesByLanguage[language]) {
+    this.textBytesByLanguage[language] = {};
+  }
+  if (!this.textBytesByLanguage[language][text]) {
+    this.textBytesByLanguage[language][text] = {};
+  }
```

Comment:
nit: simplify to `this.textBytesByLanguage.language.text = this.textBytesByLanguage.language.text || {}` (and above)